src/components/engineService.ts

```diff
@@ -200,7 +200,9 @@ export class EngineService {
 
 	public async fetchPrivateCodemods(): Promise<void> {
 		try {
-			const privateCodemods: CodemodEntry[] = [];
+			const privateCodemods: (CodemodEntry & {
+				permalink: string | null;
```

Comment:
can't we calculate the permalink on demand? IMO this shouldn't be part of the entry